follower/follower_builder.go

```diff
@@ -0,0 +1,723 @@
+	"github.com/onflow/flow-go/network/p2p/keyutils"
+	"github.com/onflow/flow-go/network/p2p/unicast"
+	"github.com/onflow/flow-go/state/protocol/events/gadgets"
+	upstream "github.com/onflow/flow-go/upstream"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/module/compliance"
+
+	"github.com/onflow/flow-go/cmd"
+	"github.com/onflow/flow-go/consensus"
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/committees"
+	"github.com/onflow/flow-go/consensus/hotstuff/notifications/pubsub"
+	hotsignature "github.com/onflow/flow-go/consensus/hotstuff/signature"
+	"github.com/onflow/flow-go/consensus/hotstuff/verification"
+	recovery "github.com/onflow/flow-go/consensus/recovery/protocol"
+	"github.com/onflow/flow-go/engine/common/follower"
+	followereng "github.com/onflow/flow-go/engine/common/follower"
+	synceng "github.com/onflow/flow-go/engine/common/synchronization"
+	"github.com/onflow/flow-go/model/encodable"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/buffer"
+	finalizer "github.com/onflow/flow-go/module/finalizer/consensus"
+	"github.com/onflow/flow-go/module/id"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/module/synchronization"
+	"github.com/onflow/flow-go/network"
+	netcache "github.com/onflow/flow-go/network/cache"
+	cborcodec "github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/validator"
+	"github.com/onflow/flow-go/state/protocol"
+	badgerState "github.com/onflow/flow-go/state/protocol/badger"
+	"github.com/onflow/flow-go/state/protocol/blocktimer"
+	storage "github.com/onflow/flow-go/storage/badger"
+)
+
+// FlowBuilder extends cmd.NodeBuilder and declares additional functions needed to bootstrap an Access node
+// These functions are shared by staked and observer builders.
+// The Staked network allows the staked nodes to communicate among themselves, while the public network allows the
+// observers and an Access node to communicate.
+//
+//                                 public network                           staked network
+//  +------------------------+
+//  | observer 1             |<--------------------------|
+//  +------------------------+                           v
+//  +------------------------+                         +----------------------+              +------------------------+
+//  | observer 2             |<----------------------->| Access Node (staked) |<------------>| All other staked Nodes |
+//  +------------------------+                         +----------------------+              +------------------------+
+//  +------------------------+                           ^
+//  | observer 3             |<--------------------------|
+//  +------------------------+
+
+type FlowBuilder interface {
```

Comment:
we probably don't need this interface since it's only used as a return type and doesn't define any changes to `cmd.NodeBuilder`